doc/src/manual/performance-tips.md

```diff
@@ -778,6 +778,7 @@ adopted by C and Python (`numpy`) among other languages. Remembering the orderin
 have significant performance effects when looping over arrays. A rule of thumb to keep in mind
 is that with column-major arrays, the first index changes most rapidly. Essentially this means
 that looping will be faster if the inner-most loop index is the first to appear in a slice expression.
+Keep in mind that accessing an array with `:` is a loop.
```

Comment:
Perhaps a bit more?  I don't want to oversell this, though, as it's not nearly as bit of a deal as scalar accesses.

```suggestion
Keep in mind that indexing an array with `:` is an implicit loop that iteratively accesses all elements within a particular dimension; it can be faster to extract columns than rows, for example.
```

Reply:
Sure, it's mostly to understand the point of the preceding paragraph and sample code.